substrate/client/network/transactions/src/lib.rs

```diff
@@ -471,7 +471,8 @@ where
 
 		debug!(target: LOG_TARGET, "Propagating transaction [{:?}]", hash);
 		if let Some(transaction) = self.transaction_pool.transaction(hash) {
-			let propagated_to = self.do_propagate_transactions(&[(hash.clone(), transaction)]);
+			let propagated_to =
+				self.do_propagate_transactions(&[(hash.clone(), Arc::new(transaction))]);
```

Comment:
Could we avoid `Arc::new` here?

I think that this `transaction` method could also provide an `Arc` reference, here:
https://github.com/paritytech/polkadot-sdk/blob/682f8cd22f5bcb76d1b98820b62be49d11deae10/substrate/client/service/src/lib.rs#L562-L567

Not sure how big that change would be, but should be doable.
Are you open to explore it?